### ISSUE TITLE
Booting GRUB4DOS from ReactOS' FreeLoader

### ISSUE THREAD

**Issue by jschwartzenberg** [chenall/grub4dos#77]
I'm trying to boot GRUB4DOS from FreeLoader, the bootloader used by ReactOS. My system has a single harddisk with a single FAT partition with ReactOS installed.

FreeLoader supports multiple ways of booting different systems. (Windows/ReactOS-like, Linux kernels, bootsectors.) It seems the bootsector approach is one that could work with GRUB4DOS on the same partition. I created an entry like this:

    [GRUB4DOS]
    BootType=BootSector
    BootSectorFile=multi(0)disk(0)rdisk(0)partition(1)\grldr.mbr

(Note that the example documentation uses BootSector instead of BootSectorFile but this is incorrect.)

In the root directory I copied both grldr.mbr and grldr.

When I boot the new entry, it loads the bootsector correctly but then this gives:

    Missing helper.

I have the impression it has difficulty locating the grldr in my root directory for some reason.

Does this approach make sense or maybe there's another way to load GRUB4DOS from FreeLoader?

**Comment by steve6375** [comment]
I would try booting to the /grldr file as a 'kernel'

[GRUB4DOS]
BootType=Linux
BootDrive=hd0
BootPartition=1
Kernel=/grldr

**Comment by jschwartzenberg** [comment]
I tried this, but it seems that FreeLoader can only load 'real' Linux kernels this way. I tried both your syntax and this one:

    [GRUB4DOS]
    BootType=Linux
    BootPath=multi(0)disk(0)rdisk(1)partition(1)
    Kernel=/grldr

**Comment by steve6375** [comment]
As a workaround, you could install grub4dos and then boot ReactOS from grub4dos?

**Comment by jschwartzenberg** [comment]
I should try that yes, but I can imagine this may not work either. I tried using a FreeDOS bootsector now and then it gives a similar issue, it says it can't find KERNEL.SYS even though it's also in the root directory.
Could it be possible that ReactOS installer formats the filesystem in a peculiar way that both GRUB4DOS and FreeDOS cannot handle?

**Comment by steve6375** [comment]
Why not boot from a grub4dos USB drive and see if you can access the files?
e.g.
find
ls (hd1,0)/
ls (hd1,1)/
rootnoverify (hd1,0)
kernel (hd1,0)/freeldr.sys
boot

**Comment by jschwartzenberg** [comment]
Yep, good idea. I just tried that (with a floppy image) and it works fine. GRUB4DOS can see all the files properly then. It can both load FreeLoader and FreeDOS without issues. It seems there is something odd about FreeLoader's bootsector support.
